test/controllers/authentications_controller_test.rb

```diff
@@ -161,33 +197,51 @@
 
     describe 'redirect_matching_account' do
       it 'must sign in an existing user through github' do
-        github_stub = stub(email: account.email, login: account.login, access_token: Faker::Lorem.word)
-        @controller.stubs(:github_api).returns(github_stub)
+        @controller.stubs(:github_api).returns(github_account_stub)
 
         get :github_callback, code: Faker::Lorem.word
 
         account.reload
         must_redirect_to account
         request.env[:clearance].current_user.id.must_equal account.id
-        account.github_verification.token.must_equal github_stub.access_token
+        account.github_verification.token.must_equal github_account_stub.access_token
+      end
+
+      it 'must sign in a verified user regardless of github restrictions' do
+        github_account_stub.stubs(:created_at).returns(Time.current)
+        @controller.stubs(:github_api).returns(github_account_stub)
+
+        get :github_callback, code: Faker::Lorem.word
+
+        request.env[:clearance].current_user.id.must_equal account.id
+      end
+
+      it 'wont sign in a non github verified user which fails github restrictions' do
+        github_account_stub.stubs(:created_at).returns(Time.current)
+        @controller.stubs(:github_api).returns(github_account_stub)
+        account.verifications.delete_all
+        FirebaseVerification.any_instance.stubs(:generate_token)
+        create(:firebase_verification, account: account)
+
+        get :github_callback, code: Faker::Lorem.word
+
+        request.env[:clearance].current_user.must_be_nil
```

Comment:
Have I missed the test case where the account is > 1.month, but there are no repositories with recognizable code?

Reply:
Yes, it is at line 104. 
https://github.com/blackducksoftware/ohloh-ui/blob/562eb5e8fd7760ed34a52ee6a5d911af478ed99d/test/controllers/authentications_controller_test.rb#L104

This uses a response recorded from authenticating a github user alex-bds.

Reply:
Thank you